build: Building depends for the `riscv64-linux-gnu` host fails

On the main branch (2aa44e608ac70ed7e56994634d8c15cf192f8390):
```
$ env HOSTS='riscv64-linux-gnu' ./contrib/guix/guix-build
...
linking ../../bin/qml
riscv64-linux-gnu-ld: /bitcoin/depends/work/build/riscv64-linux-gnu/qt/5.12.11-a191f20871a/qtdeclarative/lib/libQt5Qml.a(qv4typedarray.o): in function `.L567':
qv4typedarray.cpp:(.text._Z9atomicXorIhEyPcN3QV45ValueE[_Z9atomicXorIhEyPcN3QV45ValueE]+0x72): undefined reference to `__atomic_fetch_xor_1'
riscv64-linux-gnu-ld: /bitcoin/depends/work/build/riscv64-linux-gnu/qt/5.12.11-a191f20871a/qtdeclarative/lib/libQt5Qml.a(qv4typedarray.o): in function `.L590':
qv4typedarray.cpp:(.text._Z9atomicXorItEyPcN3QV45ValueE[_Z9atomicXorItEyPcN3QV45ValueE]+0x74): undefined reference to `__atomic_fetch_xor_2'
riscv64-linux-gnu-ld: /bitcoin/depends/work/build/riscv64-linux-gnu/qt/5.12.11-a191f20871a/qtdeclarative/lib/libQt5Qml.a(qv4typedarray.o): in function `.L613':
qv4typedarray.cpp:(.text._Z9atomicSubIhEyPcN3QV45ValueE[_Z9atomicSubIhEyPcN3QV45ValueE]+0x72): undefined reference to `__atomic_fetch_sub_1'
riscv64-linux-gnu-ld: /bitcoin/depends/work/build/riscv64-linux-gnu/qt/5.12.11-a191f20871a/qtdeclarative/lib/libQt5Qml.a(qv4typedarray.o): in function `.L636':
qv4typedarray.cpp:(.text._Z8atomicOrItEyPcN3QV45ValueE[_Z8atomicOrItEyPcN3QV45ValueE]+0x74): undefined reference to `__atomic_fetch_or_2'
riscv64-linux-gnu-ld: /bitcoin/depends/work/build/riscv64-linux-gnu/qt/5.12.11-a191f20871a/qtdeclarative/lib/libQt5Qml.a(qv4typedarray.o): in function `.L659':
qv4typedarray.cpp:(.text._Z9atomicAddIaEyPcN3QV45ValueE[_Z9atomicAddIaEyPcN3QV45ValueE]+0x72): undefined reference to `__atomic_fetch_add_1'
riscv64-linux-gnu-ld: /bitcoin/depends/work/build/riscv64-linux-gnu/qt/5.12.11-a191f20871a/qtdeclarative/lib/libQt5Qml.a(qv4typedarray.o): in function `.L682':
qv4typedarray.cpp:(.text._Z8atomicOrIaEyPcN3QV45ValueE[_Z8atomicOrIaEyPcN3QV45ValueE]+0x72): undefined reference to `__atomic_fetch_or_1'
riscv64-linux-gnu-ld: /bitcoin/depends/work/build/riscv64-linux-gnu/qt/5.12.11-a191f20871a/qtdeclarative/lib/libQt5Qml.a(qv4typedarray.o): in function `.L705':
qv4typedarray.cpp:(.text._Z14atomicExchangeIaEyPcN3QV45ValueE[_Z14atomicExchangeIaEyPcN3QV45ValueE]+0x72): undefined reference to `__atomic_exchange_1'
riscv64-linux-gnu-ld: /bitcoin/depends/work/build/riscv64-linux-gnu/qt/5.12.11-a191f20871a/qtdeclarative/lib/libQt5Qml.a(qv4typedarray.o): in function `.L728':
qv4typedarray.cpp:(.text._Z8atomicOrIsEyPcN3QV45ValueE[_Z8atomicOrIsEyPcN3QV45ValueE]+0x74): undefined reference to `__atomic_fetch_or_2'
riscv64-linux-gnu-ld: /bitcoin/depends/work/build/riscv64-linux-gnu/qt/5.12.11-a191f20871a/qtdeclarative/lib/libQt5Qml.a(qv4typedarray.o): in function `.L751':
qv4typedarray.cpp:(.text._Z9atomicXorIaEyPcN3QV45ValueE[_Z9atomicXorIaEyPcN3QV45ValueE]+0x72): undefined reference to `__atomic_fetch_xor_1'
riscv64-linux-gnu-ld: /bitcoin/depends/work/build/riscv64-linux-gnu/qt/5.12.11-a191f20871a/qtdeclarative/lib/libQt5Qml.a(qv4typedarray.o): in function `.L774':
qv4typedarray.cpp:(.text._Z9atomicAndItEyPcN3QV45ValueE[_Z9atomicAndItEyPcN3QV45ValueE]+0x74): undefined reference to `__atomic_fetch_and_2'
riscv64-linux-gnu-ld: /bitcoin/depends/work/build/riscv64-linux-gnu/qt/5.12.11-a191f20871a/qtdeclarative/lib/libQt5Qml.a(qv4typedarray.o): in function `.L797':
qv4typedarray.cpp:(.text._Z9atomicSubIaEyPcN3QV45ValueE[_Z9atomicSubIaEyPcN3QV45ValueE]+0x72): undefined reference to `__atomic_fetch_sub_1'
riscv64-linux-gnu-ld: /bitcoin/depends/work/build/riscv64-linux-gnu/qt/5.12.11-a191f20871a/qtdeclarative/lib/libQt5Qml.a(qv4typedarray.o): in function `.L842':
qv4typedarray.cpp:(.text._Z9atomicXorIsEyPcN3QV45ValueE[_Z9atomicXorIsEyPcN3QV45ValueE]+0x74): undefined reference to `__atomic_fetch_xor_2'
riscv64-linux-gnu-ld: /bitcoin/depends/work/build/riscv64-linux-gnu/qt/5.12.11-a191f20871a/qtdeclarative/lib/libQt5Qml.a(qv4typedarray.o): in function `.L929':
qv4typedarray.cpp:(.text._Z9atomicAddIhEyPcN3QV45ValueE[_Z9atomicAddIhEyPcN3QV45ValueE]+0x72): undefined reference to `__atomic_fetch_add_1'
riscv64-linux-gnu-ld: /bitcoin/depends/work/build/riscv64-linux-gnu/qt/5.12.11-a191f20871a/qtdeclarative/lib/libQt5Qml.a(qv4typedarray.o): in function `.L974':
qv4typedarray.cpp:(.text._Z14atomicExchangeIhEyPcN3QV45ValueE[_Z14atomicExchangeIhEyPcN3QV45ValueE]+0x72): undefined reference to `__atomic_exchange_1'
riscv64-linux-gnu-ld: /bitcoin/depends/work/build/riscv64-linux-gnu/qt/5.12.11-a191f20871a/qtdeclarative/lib/libQt5Qml.a(qv4typedarray.o): in function `.L997':
qv4typedarray.cpp:(.text._Z9atomicAndIaEyPcN3QV45ValueE[_Z9atomicAndIaEyPcN3QV45ValueE]+0x72): undefined reference to `__atomic_fetch_and_1'
riscv64-linux-gnu-ld: /bitcoin/depends/work/build/riscv64-linux-gnu/qt/5.12.11-a191f20871a/qtdeclarative/lib/libQt5Qml.a(qv4typedarray.o): in function `.L1020':
qv4typedarray.cpp:(.text._Z8atomicOrIhEyPcN3QV45ValueE[_Z8atomicOrIhEyPcN3QV45ValueE]+0x72): undefined reference to `__atomic_fetch_or_1'
riscv64-linux-gnu-ld: /bitcoin/depends/work/build/riscv64-linux-gnu/qt/5.12.11-a191f20871a/qtdeclarative/lib/libQt5Qml.a(qv4typedarray.o): in function `.L1064':
qv4typedarray.cpp:(.text._Z9atomicAndIhEyPcN3QV45ValueE[_Z9atomicAndIhEyPcN3QV45ValueE]+0x72): undefined reference to `__atomic_fetch_and_1'
riscv64-linux-gnu-ld: /bitcoin/depends/work/build/riscv64-linux-gnu/qt/5.12.11-a191f20871a/qtdeclarative/lib/libQt5Qml.a(qv4typedarray.o): in function `.L1109':
qv4typedarray.cpp:(.text._Z14atomicExchangeItEyPcN3QV45ValueE[_Z14atomicExchangeItEyPcN3QV45ValueE]+0x74): undefined reference to `__atomic_exchange_2'
riscv64-linux-gnu-ld: /bitcoin/depends/work/build/riscv64-linux-gnu/qt/5.12.11-a191f20871a/qtdeclarative/lib/libQt5Qml.a(qv4typedarray.o): in function `.L1132':
qv4typedarray.cpp:(.text._Z9atomicSubItEyPcN3QV45ValueE[_Z9atomicSubItEyPcN3QV45ValueE]+0x74): undefined reference to `__atomic_fetch_sub_2'
riscv64-linux-gnu-ld: /bitcoin/depends/work/build/riscv64-linux-gnu/qt/5.12.11-a191f20871a/qtdeclarative/lib/libQt5Qml.a(qv4typedarray.o): in function `.L1155':
qv4typedarray.cpp:(.text._Z9atomicAddIsEyPcN3QV45ValueE[_Z9atomicAddIsEyPcN3QV45ValueE]+0x74): undefined reference to `__atomic_fetch_add_2'
riscv64-linux-gnu-ld: /bitcoin/depends/work/build/riscv64-linux-gnu/qt/5.12.11-a191f20871a/qtdeclarative/lib/libQt5Qml.a(qv4typedarray.o): in function `.L1200':
qv4typedarray.cpp:(.text._Z14atomicExchangeIsEyPcN3QV45ValueE[_Z14atomicExchangeIsEyPcN3QV45ValueE]+0x74): undefined reference to `__atomic_exchange_2'
riscv64-linux-gnu-ld: /bitcoin/depends/work/build/riscv64-linux-gnu/qt/5.12.11-a191f20871a/qtdeclarative/lib/libQt5Qml.a(qv4typedarray.o): in function `.L1223':
qv4typedarray.cpp:(.text._Z9atomicAndIsEyPcN3QV45ValueE[_Z9atomicAndIsEyPcN3QV45ValueE]+0x74): undefined reference to `__atomic_fetch_and_2'
riscv64-linux-gnu-ld: /bitcoin/depends/work/build/riscv64-linux-gnu/qt/5.12.11-a191f20871a/qtdeclarative/lib/libQt5Qml.a(qv4typedarray.o): in function `.L1246':
qv4typedarray.cpp:(.text._Z9atomicSubIsEyPcN3QV45ValueE[_Z9atomicSubIsEyPcN3QV45ValueE]+0x74): undefined reference to `__atomic_fetch_sub_2'
riscv64-linux-gnu-ld: /bitcoin/depends/work/build/riscv64-linux-gnu/qt/5.12.11-a191f20871a/qtdeclarative/lib/libQt5Qml.a(qv4typedarray.o): in function `.L1269':
qv4typedarray.cpp:(.text._Z9atomicAddItEyPcN3QV45ValueE[_Z9atomicAddItEyPcN3QV45ValueE]+0x74): undefined reference to `__atomic_fetch_add_2'
riscv64-linux-gnu-ld: /bitcoin/depends/work/build/riscv64-linux-gnu/qt/5.12.11-a191f20871a/qtdeclarative/lib/libQt5Qml.a(qv4typedarray.o): in function `.L1539':
qv4typedarray.cpp:(.text._Z21atomicCompareExchangeItEyPcN3QV45ValueES2_[_Z21atomicCompareExchangeItEyPcN3QV45ValueES2_]+0xde): undefined reference to `__atomic_compare_exchange_2'
riscv64-linux-gnu-ld: /bitcoin/depends/work/build/riscv64-linux-gnu/qt/5.12.11-a191f20871a/qtdeclarative/lib/libQt5Qml.a(qv4typedarray.o): in function `.L1584':
qv4typedarray.cpp:(.text._Z21atomicCompareExchangeIaEyPcN3QV45ValueES2_[_Z21atomicCompareExchangeIaEyPcN3QV45ValueES2_]+0xe2): undefined reference to `__atomic_compare_exchange_1'
riscv64-linux-gnu-ld: /bitcoin/depends/work/build/riscv64-linux-gnu/qt/5.12.11-a191f20871a/qtdeclarative/lib/libQt5Qml.a(qv4typedarray.o): in function `.L1629':
qv4typedarray.cpp:(.text._Z21atomicCompareExchangeIhEyPcN3QV45ValueES2_[_Z21atomicCompareExchangeIhEyPcN3QV45ValueES2_]+0xde): undefined reference to `__atomic_compare_exchange_1'
riscv64-linux-gnu-ld: /bitcoin/depends/work/build/riscv64-linux-gnu/qt/5.12.11-a191f20871a/qtdeclarative/lib/libQt5Qml.a(qv4typedarray.o): in function `.L1674':
qv4typedarray.cpp:(.text._Z21atomicCompareExchangeIsEyPcN3QV45ValueES2_[_Z21atomicCompareExchangeIsEyPcN3QV45ValueES2_]+0xe2): undefined reference to `__atomic_compare_exchange_2'
collect2: error: ld returned 1 exit status
...
```

--- a/depends/packages/qt.mk
+++ b/depends/packages/qt.mk
@@ -13,7 +13,7 @@ $(package)_patches += support_new_android_ndks.patch fix_android_jni_static.patc
 $(package)_patches+= no_sdk_version_check.patch
 $(package)_patches+= fix_lib_paths.patch fix_android_pch.patch
 $(package)_patches+= qtbase-moc-ignore-gcc-macro.patch fix_limits_header.patch
-$(package)_patches += fix_qml_python.patch
+$(package)_patches += fix_qml_python.patch riscv_detection.patch
 
 $(package)_qtdeclarative_file_name = qtdeclarative-$($(package)_suffix)
 $(package)_qtdeclarative_sha256_hash = 1267e029abc8424424c419bc1681db069ec76e51270cc220994e0f442c9f78d3
@@ -267,6 +267,7 @@ define $(package)_preprocess_cmds
   patch -p1 -i $($(package)_patch_dir)/qtbase-moc-ignore-gcc-macro.patch && \
   patch -p1 -i $($(package)_patch_dir)/fix_limits_header.patch && \
   patch -p1 -i $($(package)_patch_dir)/fix_qml_python.patch && \
+  patch -p1 -i $($(package)_patch_dir)/riscv_detection.patch && \
   mkdir -p qtbase/mkspecs/macx-clang-linux &&\
   cp -f qtbase/mkspecs/macx-clang/qplatformdefs.h qtbase/mkspecs/macx-clang-linux/ &&\
   cp -f $($(package)_patch_dir)/mac-qmake.conf qtbase/mkspecs/macx-clang-linux/qmake.conf && \

--- a/depends/packages/qt.mk
+++ b/depends/packages/qt.mk
@@ -13,7 +13,7 @@ $(package)_patches += support_new_android_ndks.patch fix_android_jni_static.patc
 $(package)_patches+= no_sdk_version_check.patch
 $(package)_patches+= fix_lib_paths.patch fix_android_pch.patch
 $(package)_patches+= qtbase-moc-ignore-gcc-macro.patch fix_limits_header.patch
-$(package)_patches += fix_qml_python.patch riscv_detection.patch
+$(package)_patches += fix_qml_python.patch riscv_detection.patch fix_riscv_atomic.patch
 
 $(package)_qtdeclarative_file_name = qtdeclarative-$($(package)_suffix)
 $(package)_qtdeclarative_sha256_hash = 1267e029abc8424424c419bc1681db069ec76e51270cc220994e0f442c9f78d3
@@ -268,6 +268,7 @@ define $(package)_preprocess_cmds
   patch -p1 -i $($(package)_patch_dir)/fix_limits_header.patch && \
   patch -p1 -i $($(package)_patch_dir)/fix_qml_python.patch && \
   patch -p1 -i $($(package)_patch_dir)/riscv_detection.patch && \
+  patch -p1 -i $($(package)_patch_dir)/fix_riscv_atomic.patch && \
   mkdir -p qtbase/mkspecs/macx-clang-linux &&\
   cp -f qtbase/mkspecs/macx-clang/qplatformdefs.h qtbase/mkspecs/macx-clang-linux/ &&\
   cp -f $($(package)_patch_dir)/mac-qmake.conf qtbase/mkspecs/macx-clang-linux/qmake.conf && \

--- a/depends/patches/qt/fix_riscv_atomic.patch
+++ b/depends/patches/qt/fix_riscv_atomic.patch
@@ -1,0 +1,17 @@
+A workaround for error when building for riscv64.
+
+Upstream report: https://bugreports.qt.io/browse/QTBUG-84580
+
+Patch concept: https://bugs.gentoo.org/790689
+
+
+--- old/qtdeclarative/src/qml/qml.pro
++++ new/qtdeclarative/src/qml/qml.pro
+@@ -19,6 +19,7 @@ solaris-cc*:QMAKE_CXXFLAGS_RELEASE -= -O2
+ 
+ # Ensure this gcc optimization is switched off for mips platforms to avoid trouble with JIT.
+ gcc:isEqual(QT_ARCH, "mips"): QMAKE_CXXFLAGS += -fno-reorder-blocks
++gcc:isEqual(QT_ARCH, "riscv64"): LIBS += -latomic
+ 
+ DEFINES += QT_NO_FOREACH
+ 

--- a/depends/patches/qt/riscv_detection.patch
+++ b/depends/patches/qt/riscv_detection.patch
@@ -1,0 +1,41 @@
+Add RISC-V detection
+
+Upstream commit:
+ - Qt 5.14: 9a6a84731131b205f74b10f866ae212e0895bd4a
+
+--- old/qtbase/src/corelib/global/archdetect.cpp
++++ new/qtbase/src/corelib/global/archdetect.cpp
+@@ -67,6 +67,10 @@
+ #  define ARCH_PROCESSOR "power"
+ #elif defined(Q_PROCESSOR_POWER_64)
+ #  define ARCH_PROCESSOR "power64"
++#elif defined(Q_PROCESSOR_RISCV_32)
++#  define ARCH_PROCESSOR "riscv32"
++#elif defined(Q_PROCESSOR_RISCV_64)
++#  define ARCH_PROCESSOR "riscv64"
+ #elif defined(Q_PROCESSOR_S390_X)
+ #  define ARCH_PROCESSOR "s390x"
+ #elif defined(Q_PROCESSOR_S390)
+
+--- old/qtbase/src/corelib/global/qprocessordetection.h
++++ old/qtbase/src/corelib/global/qprocessordetection.h
+@@ -282,6 +282,19 @@
+ // Q_BYTE_ORDER not defined, use endianness auto-detection
+ 
++ /*
++    RISC-V family, known variants: 32- and 64-bit
++    RISC-V is little-endian.
++*/
++#elif defined(__riscv)
++#  define Q_PROCESSOR_RISCV
++#  if __riscv_xlen == 64
++#    define Q_PROCESSOR_RISCV_64
++#  else
++#    define Q_PROCESSOR_RISCV_32
++#  endif
++#  define Q_BYTE_ORDER Q_LITTLE_ENDIAN
++
+ /*
+     S390 family, known variant: S390X (64-bit)
+ 
+     S390 is big-endian.

